src/App.jsx

```diff
@@ -14,45 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_METHOD_ALPHABET = 'Alphabet';
+const SORT_METHOD_LENGTH = 'length';
+
+function getPrepearedGoods(goods, sortMethod, reverse) {
+  const prepearedGoods = [...goods];
+
+  if (sortMethod) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortMethod) {
+        case SORT_METHOD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_METHOD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [reverse, setReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPrepearedGoods(goodsFromServer, sortMethod, reverse);
+
+  const toggleReverse = () => {
+    setReverse(!reverse);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortMethod !== SORT_METHOD_ALPHABET,
+          })}
+          onClick={() => {
+            setSortMethod(SORT_METHOD_ALPHABET);
+            setReverse(false); // Reset reverse when changing sort method
+          }}
```

Comment:
- it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here
- remove all comments